pinot-query-planner/src/main/java/org/apache/pinot/query/QueryEnvironment.java

```diff
@@ -391,7 +411,17 @@ private RelNode optimize(RelRoot relRoot, PlannerContext plannerContext) {
     try {
       RelOptPlanner optPlanner = plannerContext.getRelOptPlanner();
       optPlanner.setRoot(relRoot.rel);
+      RuleTimingPlannerListener listener = null;
+      if (plannerContext.getOptions()
+          .containsKey(CommonConstants.Broker.Request.QueryOptionKey.PROFILE_RULE_OPTIMIZER)) {
```

Comment:
(minor) Extract this into `QueryOptionsUtils`. `containsKey()` is not enough. We should check the value using `Boolean.parseBoolean()`